clusterctl/clusterdeployer/clusterdeployer.go

```diff
@@ -388,13 +451,45 @@ func pivot(from, to ClusterClient) error {
 		machine.SetResourceVersion("")
 		err = to.CreateMachineObjects([]*clusterv1.Machine{machine})
 		if err != nil {
-			return err
+			return fmt.Errorf("error moving Machine '%v': %v", machine.GetName(), err)
 		}
 		glog.Infof("Moved Machine '%s'", machine.GetName())
 	}
 	return nil
 }
 
+func deleteObjects(client ClusterClient) error {
```

Comment:
Wondering two things 1) if I get an `error` can I call it again and 2) from a readability perspective would it be better to split it up into 3 helper functions: (`DeleteMachineDeploymentObjects`, `DeleteMachineSetObjects`, `DeleteMachineObjects`).